Classes/nameable.rb

```diff
@@ -0,0 +1,26 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+end
```

Comment:
- [ ] The class `Decorator` should have `correct_name` method that returns the result of the correct_name method of the @nameable is implemented.

```ruby
class Decorator < Nameable
    
    ......
    ......
  

    def correct_name
      @nameable.correct_name
    end
end
```